project/driver_controls/src/main.c

```diff
@@ -0,0 +1,131 @@
+#include <stdio.h>
+#include <stdint.h>
+
+#include "adc.h"
+#include "input_interrupt.h"
+#include "driver_state.h"
+#include "soft_timer.h"
+#include "driver_device.h"
+
+#include "power_state.h"
+#include "pedal_state.h"
+#include "direction_state.h"
+#include "turn_signal_state.h"
+#include "hazard_light_state.h"
+#include "mechanical_brake_state.h"
+
+#define INPUT_DEVICES 16
```

Comment:
You could use `inputs[]` and `SIZEOF_ARRAY(inputs)` instead of defining the value here. Since the input array is defined on the stack and not passed as a pointer, the compiler still knows how big it actually is so it can calculate the size at compile-time.